src/Processors/Formats/Impl/ArrowFieldIndexUtil.h

```diff
@@ -21,9 +21,7 @@ namespace ErrorCodes
 {
     extern const int LOGICAL_ERROR;
 }
-/// For ORC format, index_nested_type = true, a nested type takes one index count. And the
-/// the start index for ORC format should be 1, since index 0 indicates to select all columns.
-template<bool index_nested_type>
+
 class ArrowFieldIndexUtil
```

Comment:
why are we removing ORC format?

Reply:
Because in the new arrow release the behaviour of ORC indexes changed, this code is not applicable to ORC anymore. I added new logic for ORC indexes in ORCBlockInputFormat.cpp

Reply:
So is the struct field pruning  lost?

Reply:
I guess yes. But there is a chance that I missed things up and we still can do it. Recently ORC input format was reimplemented using orc c++ library without arrow adapter. I think I can implement it there and also I have an idea how to optimize reading subcolumns. I will try

Reply:
Ah, went through c++ orc library api and they allow only indexes for top level struct (and it's used), but not for nested structs. To optimize it, we will need to patch orc library